csi/csi.go

```diff
@@ -155,6 +173,66 @@ func (s *OsdCsiServer) getConn() (*grpc.ClientConn, error) {
 	return s.conn, nil
 }
 
+func (s *OsdCsiServer) getRemoteConn() (*grpc.ClientConn, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	// Get all nodes and sort them
+	nodesResp, err := s.cluster.Enumerate()
+	if err != nil {
+		return nil, err
+	}
+	sort.Slice(nodesResp.Nodes, func(i, j int) bool {
+		return nodesResp.Nodes[i].Id < nodesResp.Nodes[j].Id
+	})
+
+	// Clean up connections for missing node
+	nodesMap := make(map[string]bool)
+	for _, node := range nodesResp.Nodes {
+		nodesMap[node.MgmtIp] = true
+	}
+	for ip := range s.connMap {
+		// If key in connmap is not in current nodes, remove it
+		if ok := nodesMap[ip]; !ok {
+			delete(s.connMap, ip)
```

Comment:
We should close the grpc connection as well? We have seen memory leaks where not closing a connection would result into leaking goroutines.